src/metabase/models/interface.clj

```diff
@@ -514,12 +529,14 @@
     (not (:created_at obj)) (assoc :created_at (now))))
 
 (defn- add-updated-at-timestamp [obj]
-  ;; don't stomp on `:updated_at` if it's already explicitly specified.
-  (let [changes-already-include-updated-at? (if (t2/instance? obj)
-                                              (:updated_at (t2/changes obj))
-                                              (:updated_at obj))]
+  (let [changed-fields (set (keys (if (t2/instance obj)
+                                    (t2/changes obj)
+                                    (:updated_at obj))))
+        ; don't stomp on `:updated_at` if it's already explicitly specified.
+        changes-already-include-updated-at? (some #{:updated_at} changed-fields)
+        has-non-ignored-fields? (seq (set/difference changed-fields (non-timestamped-fields obj)))]
     (cond-> obj
-      (not changes-already-include-updated-at?) (assoc :updated_at (now)))))
+      (or (empty? changed-fields) (and has-non-ignored-fields? (not changes-already-include-updated-at?))) (assoc :updated_at (now)))))
```

Comment:
maybe it would help readability to move the logic to the let block a `should-set-updated-at?` binding.